micromagneticmodel/dynamics/zhangli.py

```diff
@@ -1,15 +1,53 @@
 import collections
+import numbers
 
 import discretisedfield as df
+import numpy as np
 import ubermagutil as uu
 import ubermagutil.typesystem as ts
 
 from .dynamicsterm import DynamicsTerm
 
 
+class Scalar_Vector3(ts.Descriptor):
```

Comment:
Shouldn't we put this with the type system to be consistent?

Reply:
We decided not to because the type is rather specific and conveying the meaning in a somewhat short name near impossible.